README.md

```diff
@@ -1,707 +1,1799 @@
+
+interface RootState {
+  todos: { id: number; completed: boolean }[]
+  alerts: { id: number; read: boolean }[]
+}
+
+const state: RootState = {
+  todos: [
+    { id: 0, completed: false },
+    { id: 1, completed: true }
+  ],
+  alerts: [
+    { id: 0, read: false },
+    { id: 1, read: true }
+  ]
+}
+
+// This:
+const structuredSelector = createStructuredSelector(
+  {
+    allTodos: (state: RootState) => state.todos,
+    allAlerts: (state: RootState) => state.alerts,
+    selectedTodo: (state: RootState, id: number) => state.todos[id]
+  },
+  createSelector
+)
+
+// Is essentially the same as this:
+const selector = createSelector(
+  [
+    (state: RootState) => state.todos,
+    (state: RootState) => state.alerts,
+    (state: RootState, id: number) => state.todos[id]
+  ],
+  (allTodos, allAlerts, selectedTodo) => {
+    return {
+      allTodos,
+      allAlerts,
+      selectedTodo
+    }
+  }
+)
+```
+
+##### Simple Use Case
+
+```ts
+const selectA = state => state.a
+const selectB = state => state.b
+
+// The result function in the following selector
+// is simply building an object from the input selectors
+const structuredSelector = createSelector(selectA, selectB, (a, b) => ({
+  a,
+  b
+}))
+
+const result = structuredSelector({ a: 1, b: 2 }) // will produce { x: 1, y: 2 }
+```
+
+</details>
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+<a id="debuggingtools"></a>
+
+## Debugging Tools
+
+<a id="developmentonlychecks"></a>
+
+<details><summary><b>Development-only Checks - (<code>Since</code> 5.0.0)</b></summary>
+
+<a id="inputstabilitycheck"></a>
+
+#### `inputStabilityCheck`
+
+Due to how [`Cascading Memoization`][**_Cascading Memoization_**] works in `Reselect`, it is crucial that your [`input selectors`] do not return a new reference on each run. If an [`input selector`][`input selectors`] always returns a new reference, like
+
+```ts
+state => ({ a: state.a, b: state.b })
+```
+
+or
+
+```ts
+state => state.todos.map(todo => todo.id)
+```
+
+that will cause the selector to never memoize properly.
+Since this is a common mistake, we've added a development mode check to catch this. By default, [`createSelector`] will now run the [`input selectors`] twice during the first call to the selector. If the result appears to be different for the same call, it will log a warning with the arguments and the two different sets of extracted input values.
+
+```ts
+type StabilityCheckFrequency = 'always' | 'once' | 'never'
+```
+
+| Possible Values | Description                                     |
+| :-------------- | :---------------------------------------------- |
+| `once`          | Run only the first time the selector is called. |
+| `always`        | Run every time the selector is called.          |
+| `never`         | Never run the `input stability check`.          |
+
+> [!IMPORTANT]
+> The `input stability check` is automatically disabled in production environments.
```

Comment:
:hand: don't need to format words like these.

```suggestion
> The input stability check is automatically disabled in production environments.
```

Reply:
Yes sire!